### PR TITLE
feat: check offset arithmetic overflow

### PR DESCRIPTION
**Pull request by doitian**

### CODE REVIEW

File: tests/offset_overflow_test.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+pub mod common;
+
+use common::table_field_generated::example as reader;
+use common::table_field_generated_verifier::example as verifier;
+use common::table_field_generated_verifier::get_root;
+
+#[test]
+fn test_soffset_overflow() {
```
Comment (doitian) [inline tests/offset_overflow_test.rs:8]: Only add a test for overflow when following soffset.

It is hard to test other overflows since they require a huge buffer.
